File: GameProfile/Front/angular/src/app/services/models/game.ts
export interface Game {
    id: string
    title: string
    releaseDate: string
    headerImage: string
    nsfw: boolean
    description: string
    screenshots: screenshot[]
    genres: genre[]
    publishers: publisher[]
    developers: developer[]
    shopsLinkBuyGame: shopsLinkBuyGame[]
    profileHasGames: any[]
    achievementsCount: number
  }

  export interface GameForBrowse
  {
    id: string
    title: string
    releaseDate: string
    headerImage: string
    genres: genre[]
    publishers: publisher[]
    developers: developer[]
    review: review[]
    status: StatusGameProgressions
  }

  export interface review
  {
    site: number
    score:number
  }
  
  export interface screenshot {
    uri: string
  }
  
  export interface genre {
    gameString: string
  }
  
  export interface publisher {
    gameString: string
  }
  
  export interface developer {
    gameString: string
  }
  
  export interface shopsLinkBuyGame {
    uri: string
  }

  export interface GamePut {
    title: string
    releaseDate: string
    headerImage: string
    nsfw: boolean
    description: string
    screenshots: screenshot[]
    genres: genre[]
    publishers: publisher[]
    developers: developer[]
    shopsLinkBuyGame: shopsLinkBuyGame[]
    achievementsCount: number
  }
  export class GameClass{
    id!: string
    title!: string
    releaseDate!: string
    headerImage!: string
    nsfw!: boolean
    description!: string
    screenshots!: screenshot[]
    genres!: genre[]
    publishers!: publisher[]
    developers!: developer[]
    shopsLinkBuyGame!: shopsLinkBuyGame[]
    achievementsCount!: number
  }

  export interface GameForProfile{
    id:string;
    title:string;
    headerImage:string;
    hours:number;
    statusGame: StatusGameProgressions;
  }
  
  export enum StatusGameProgressions
  {
      Playing=1,
      Completed=2,
      Dropped=3,
      Planned=4,
      NONE=0
  }

File: GameProfile/Back/Monolith/GameProfile.Domain/Entities/GameEntites/Game.cs
using GameProfile.Domain.Entities.Forum;
using GameProfile.Domain.Entities.ProfileEntites;
using GameProfile.Domain.ValueObjects.Game;

namespace GameProfile.Domain.Entities.GameEntites
{
    public sealed class Game : Entity
    {
        #region Constructors
        public Game(Guid id,
                    string title,
                    DateTime releaseDate,
                    Uri headerImage,
                    Uri backgroundImage,
                    bool nsfw,
                    string description,
                    ICollection<StringForGame> developers,
                    ICollection<StringForGame> publishers,
                    ICollection<StringForGame> genres,
                    ICollection<StringForGame> tags,
                    ICollection<UriForGame> screenshots,
                    ICollection<UriForGame> shopsLinkBuyGame,
                    ICollection<Review> reviews,
                    int achievementsCount) : this(id, title, releaseDate, headerImage, nsfw, description, achievementsCount, backgroundImage)
        {
            Developers = developers;
            Publishers = publishers;
            Genres = genres;
            Tags = tags;
            Screenshots = screenshots;
            ShopsLinkBuyGame = shopsLinkBuyGame;
            Reviews = reviews;
        }

        /// <summary>
        /// EF constructor
        /// </summary>
        
        private Game(Guid id,
                    string title,
                    DateTime releaseDate,
                    Uri headerImage,
                    bool nsfw,
                    string description,
                    int achievementsCount,
                    Uri backgroundImage) : base(id)
        {
            Title = title;
            ReleaseDate = releaseDate;
            HeaderImage = headerImage;
            Nsfw = nsfw;
            Description = description;
            AchievementsCount = achievementsCount;
            BackgroundImage = backgroundImage;
        }
        #endregion
        //I didn't do read only collection because it's a lot of unnecessary code for each collection as the project is small it's not reasonable
        public string Title { get; private set; }

        public DateTime ReleaseDate { get; private set; }

        public Uri HeaderImage { get; private set; }

        public Uri BackgroundImage { get; private set; }

        public bool Nsfw { get; private set; }

        public string Description { get; private set; }

        public ICollection<UriForGame>? Screenshots { get; private set; }

        public ICollection<StringForGame>? Genres { get; private set; }

        public ICollection<StringForGame>? Publishers { get; private set; }

        public ICollection<StringForGame>? Developers { get; private set; }

        public ICollection<UriForGame>? ShopsLinkBuyGame { get; private set; }

        public ICollection<Review>? Reviews { get; private set; }

        public ICollection<StringForGame>? Tags { get; private set; }

        public int AchievementsCount { get; private set; }
        //reference propertys ef core
        public ICollection<ProfileHasGames>? ProfileHasGames { get; set; }

        public ICollection<Post>? Posts { get; set; }

        public ICollection<GameHasRatingFromProfile>? GameHasRatingFromProfiles { get; set; }

        public ICollection<GameHasComments>? GameHasComments { get; set; }

        public ICollection<Ranks>? Ranks { get; set;}

    }
}
